src/kibana/plugins/dashboard/index.js

```diff
@@ -56,6 +56,14 @@ define(function (require) {
         });
 
         var dash = $scope.dash = $route.current.locals.dash;
+
+        // I don't like this, but not sure how else to tell if this is a fresh load of the
+        // saved dashboard object, so looking for _a in the URL. Ideas?
+        if (dash.timeRestore && dash.timeTo && dash.timeFrom && !$routeParams._a) {
```

Comment:
As you pointed out, checking for `_a` is a little dangerous - it's makes assumptions about how AppState works. It's unlikely to cause problems as it's unlikely to change, but it's still not a good idea.

To me, it seems like you want to offload this to the AppState service somehow though. My immediate thought is keeping a reference to what the AppState in the URL was *before* defaults were injected, and making that "original" state available from the service, but I'm not convinced that's the answer.